packages/solid-js/src/createFireproof.tsx

```diff
@@ -0,0 +1,196 @@
+ *
+ * // As global databases -- can put these in a file and import them where you need them
+ * export const FireproofDB = createFireproof();
+ * export const AwesomeDB = createFireproof("AwesomeDB");
+ * ```
+ *
+ */
+export function createFireproof(dbName?: string, config: ConfigOpts = {}): CreateFireproof {
+  // The database connection is cached, so subsequent calls to fireproof with the same name will
+  // return the same database object. This makes it safe to invoke the getter function many times
+  // without needing to wrap it with createMemo. An added perk of not needing createMemo is this
+  // allows use of this hook at the global scope without needing to wrap it with createRoot from SolidJS.
+  const database = () => fireproof(dbName || "FireproofDB", config);
+
+  function createDocument<T extends DocRecord<T>>(initialDocFn: Accessor<Doc<T>>): CreateDocumentResult<T> {
+    const [doc, setDoc] = createSignal(initialDocFn());
+    const initialDocId = () => initialDocFn()._id;
+
+    // Memoize the docId to re-run dependent effects ONLY when the _id value actually changes
+    const docId = createMemo(() => doc()._id);
+
+    const updateDoc: UpdateDocFn<T> = (newDoc, options = { replace: false }) => {
+      setDoc(
+        !newDoc
+          ? initialDocFn
+          : (prevDoc) => (options.replace ? (newDoc as Doc<T>) : (deepmerge(prevDoc, newDoc) as Doc<T>))
+      );
+    };
+
+    const saveDoc: StoreDocFn<T> = async (existingDoc) => {
+      const response = await database().put(existingDoc ?? doc());
+      if (!existingDoc && !doc()._id) setDoc((d) => ({ ...d, _id: response.id }));
+      return response;
+    };
+
+    const refreshDoc = async (db: Database, docId = "") => {
+      // TODO: Add option for MVCC (Multi-version concurrency control) checks
+      // https://use-fireproof.com/docs/database-api/documents/#multi-version-concurrency-control-mvcc-available-in-alpha-coming-soon-in-beta
+      const storedDoc = await db.get<T>(docId).catch(initialDocFn);
+      setDoc(() => storedDoc);
+    };
+
+    createEffect(() => {
+      const subscriptionId = docId();
+      if (!subscriptionId) return;
+
+      const db = database();
+
+      const unsubscribe = db.subscribe(async (updatedDocs) => {
+        if (updatedDocs.find((c) => c._id === subscriptionId)) {
+          await refreshDoc(db, subscriptionId);
+        }
+      });
+
+      onCleanup(() => {
+        unsubscribe();
+      });
+    });
+
+    createEffect(() => {
+      void refreshDoc(database(), initialDocId());
+    });
+
+    return [doc, updateDoc, saveDoc];
+  }
+
+  function createLiveQuery<T extends DocRecord<T>>(
+    strOrFn: string | MapFn,
+    query = {},
+    initialRows: IndexRow<T>[] = []
+  ) {
+    // TODO: Explore using a store instead of a signal for more efficient updates
+    const [result, setResult] = createSignal({
+      docs: initialRows.map((r) => r.doc as Doc<T>),
+      rows: initialRows,
+    });
```

Comment:
Only have this one TODO here which I'll revisit at a later time. I think it'll be interesting to explore making this use [createStore](https://docs.solidjs.com/references/api-reference/stores/using-stores) and how we can support more fine grained updated.

But do not want to bite off more than necessary for this first API pass. I think this is definitely good enough and that store implementation would just be an optimization/refinement